### PR TITLE
Images by tag

### LINKED ISSUE #11: Images by Tag
#### As a user I want to view all images associated with a tag.

Acceptance criteria:
- [x] When I click on a tag, I see a filtered list of only the images that have
  the tag I clicked on.
- [x] The tag name (not its ID) is used to perform the filtering.

Discussion topics:
- [x] Where should we implement this functionality?
- [x] What happens if a user tries to manually filter by a nonexistent tag?

Dependencies:
- Image Tags

### PR DESCRIPTION
**Pull request by erikperkins**
Fixes #11.

Filter images by clickable tags

### CODE REVIEW

File: test/controllers/images_controller_test.rb [hunk 1]
```diff
@@ -12,6 +13,11 @@ def setup
     assert_select 'img', 3
   end
 
+  test 'tag parameter filters images' do
+    get :index, { tag: 'logic' }
+    assert_select 'img', 1
```
Comment (bboe) [inline test/controllers/images_controller_test.rb]: What status code should this return?

File: test/controllers/images_controller_test.rb [hunk 2]
```diff
@@ -12,6 +13,11 @@ def setup
...
+  end
+
   test 'get new' do
     get :new
```
Comment (bboe) [inline test/controllers/images_controller_test.rb]: What should happen if there are no images associated with the tag? Or what if tag is an array?

**Comment by bboe** [comment]
@erikperkins Great work overall. I really like how the two commits are separated. Just a few minor test related things to address.

File: test/controllers/images_controller_test.rb (hunk 2 above)
Comment (erikperkins) [inline test/controllers/images_controller_test.rb]: Addressed in 9325b6da459ee290fb4bcc22824fe91c7364bdae

File: test/controllers/images_controller_test.rb (hunk 1 above)
Comment (erikperkins) [inline test/controllers/images_controller_test.rb]: Addressed in 9325b6da459ee290fb4bcc22824fe91c7364bdae

**Comment by bboe** [comment]
Looks good! Please squash the first and last commit together and merge. :+1:

**Pull request merged**